zerver/actions/user_groups.py

```diff
@@ -301,6 +301,32 @@ def bulk_add_members_to_user_group(
     do_send_user_group_members_update_event("add_members", user_group, user_profile_ids)
 
 
+@transaction.atomic(savepoint=False)
+def bulk_add_member_to_user_groups(
+    user_groups: List[UserGroup], user_profile: UserProfile, *, acting_user: Optional[UserProfile]
+) -> None:
+    memberships = [
+        UserGroupMembership(user_group_id=user_group.id, user_profile_id=user_profile.id)
+        for user_group in user_groups
+    ]
+    UserGroupMembership.objects.bulk_create(memberships)
+    now = timezone_now()
+    RealmAuditLog.objects.bulk_create(
+        RealmAuditLog(
+            realm=user_profile.realm,
+            modified_user_id=user_profile.id,
+            modified_user_group=user_group,
+            event_type=RealmAuditLog.USER_GROUP_DIRECT_USER_MEMBERSHIP_ADDED,
+            event_time=now,
+            acting_user=acting_user,
+        )
+        for user_group in user_groups
+    )
+
+    for user_group in user_groups:
+        do_send_user_group_members_update_event("add_members", user_group, [user_profile.id])
```

Comment:
I think the ideal implementation for this would be to refactor the existing  `bulk_add_members_to_user_group` to support multiple user groups. We can do an `assert len(user_groups) == 1 or len(user_profile_ids) == 1)` if we don't want to think about the quadratic work situations; but I think it should be a pretty easy refactor and then we don't have to duplicate all this code.

Reply:
Opened https://github.com/zulip/zulip/pull/26925 to address this.